dnf-docker-test/features/steps/test_behave.py

```diff
@@ -1,106 +1,69 @@
-        map(_right_decorator, list(sb - sa)))
-
-
-def package_version_lists(pkg, list_ver):
-    """ Select package versions """
-    found_pkgs = [x for x in list_ver if x.startswith(pkg)]
-    assert len(found_pkgs) == 1
-    return str(found_pkgs[0])
+    sack = dnf.Base().fill_sack(load_available_repos=False)
+    list_sack = list(sack.query().installed())
+    assert len(list_sack) == len(comparerpmver)
+    comparerpmver_remove = list(comparerpmver)
+    for pkg in comparerpmver:
+        subj = dnf.subject.Subject(pkg)
+        candidate = subj.get_best_query(sack)
+        assert len(candidate) == 1
+        list_sack.remove(candidate[0])
+        comparerpmver_remove.remove(pkg)
+    assert not list_sack
+    assert not comparerpmver_remove
+    return sack
 
 
-def package_absence(pkg, list_ver):
-    """ Select package versions """
-    found_pkgs = [x for x in list_ver if re.search('^' + pkg, x)]
-    assert len(found_pkgs) == 0
-    return None
+def diff_query_lists(sack_a, sack_b):
+    """ Computes diff between sack_a and sack_b """
+    sa, sb = list(sack_a.query().installed()), list(sack_b.query().installed())
+    rem = [x for x in sa if x not in sb]
+    inst = [x for x in sb if x not in sa]
+    return rem, inst
 
 
-def execute_dnf_command(dnf_command_version, cmd, reponame):
-    """ Execute DNF command with default flags and the specified `reponame` enabled """
-    flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
-    subprocess.check_call([dnf_command_version] + flags + cmd, stdout=subprocess.PIPE)
-    return sleep(1)
+def ensure_path_exist(path):
+    """Ensure that os path exist. If doesn't exist it create the path."""
+    if not os.path.exists(path):
+        os.makedirs(path)
 
 
-def execute_rpm_command(pkg, action):
-    """ Execute given action over specified pkg(s) """
-    if not isinstance(pkg, list):
-        pkg = [pkg]
-    if action == "remove":
-        rpm_command = RPM_ERASE_FLAGS
-    elif action == "install":
-        rpm_command = RPM_INSTALL_FLAGS
-        pkg = decorate_rpm_packages(pkg)
-    subprocess.check_call(['rpm'] + rpm_command + pkg, stdout=subprocess.PIPE)
-    return sleep(1)
+def package_finder(sack, package_string, package_count=1):
+    """
+    @param sack: sack with packages
+    @param package_string: package description in nevra format
+    @return: name of package
 
+    """
+    subj = dnf.subject.Subject(package_string)
+    candidate = subj.get_best_query(sack)
+    assert len(candidate) == package_count, 'The number of found packages {} (packages - {}) for string "{}" differs ' \
+                                            'from expected {}'.format(
+        len(candidate), ' '.join([str(pkg) for pkg in candidate]), package_string, package_count)
 
-def piecewise_compare(a, b):
-    """ Check if the two sequences are identical regardless of ordering """
-    return sorted(a) == sorted(b)
+    if package_count == 1:
+        return candidate[0]
```

Comment:
Yes, I do use 0 at the present time. But theoretically it can be used for installonly packages with higher number than 1.  If there will be usage for higher number, I will add return for it if I will know, what should be returned.